docs/README.md

```diff
@@ -293,6 +293,19 @@ To add a new input test with a sample P4 code file (under `testdata/p4_16_sample
   * Use `ordered_map` and `ordered_set` when you need to iterate;
     they provide deterministic iterators.
 
+## Naming conventions
+
+P4 provides a rich assortment of types. Base types include bit-strings, numbers, and errors. There are also built-in types for representing constructs such as parsers, pipelines, actions, and tables. Users can construct new types based on these: structures, enumerations, headers, header stacks, header unions, etc.
+
+In this document we adopt the following conventions:
+
+- Built-in types are written with lowercase characters—e.g., `int<20>`,
+- User-defined types are capitalized—e.g., `IPv4Address`,
+- Type variables are always uppercase—e.g., parser `P<H, IH>(...)`,
+- Variables are uncapitalized— e.g., `ipv4header`,
```

Comment:
This convention can make reading variable names difficult 

example - 

`maxshiftwidth`   for  `maxShiftWidth` 
https://github.com/p4lang/p4c/blob/f8df27f677a2eedac6357139f8d40828763af75f/backends/bmv2/common/lower.h#L33

`typeargnum`  for `typeArgNum`
https://github.com/p4lang/p4c/blob/f8df27f677a2eedac6357139f8d40828763af75f/backends/dpdk/dpdkContext.cpp#L128

Reply:
Reference docs for this P4 naming conventions https://p4.org/p4-spec/docs/P4-16-v1.0.0-spec.html#sec-naming-conventions , if that's the case than this info has to be updated. @AdarshRawat1

Reply:
@fruffy Does this variable naming convention need to be updated?

Reply:
Example for first criteria :
https://github.com/p4lang/p4c/blob/f8df27f677a2eedac6357139f8d40828763af75f/backends/dpdk/dpdkArch.cpp#L1502
Since the variable used here is not capitalized. Accordingly, the rule for naming variables would be that we could only have one word with an uncapitalized first character, and the other criterion would be meeting the first condition and beginning another meaningful term with a capital character.

Example for second criteria :
https://github.com/p4lang/p4c/blob/f8df27f677a2eedac6357139f8d40828763af75f/backends/dpdk/dpdkArch.cpp#L1503

Reply:
There are two things going on here. 

1) The naming conventions defined in the P4 specifications apply to P4 programs, not the C++ code of the compiler.

2) For the compiler, we have not decided on a consistent style yet. Currently, each back end has their own style because they are used by different teams. Google style is different from LLVM style is different from Intel or Java style. There is actually an issue which we need to resolve: https://github.com/p4lang/p4c/issues/4404
I have a PR open that defines the style but it is not yet complete: https://github.com/p4lang/p4c/pull/4410
I might work on this next week. 

FYI, there are certain names for these casing styles:
- CamelCase
- camelBack
- snake_case
- kebab-case

Reply:
> There are two things going on here.
> ....... complete: [Implement the Google variable naming style more faithfully. #4410](https://github.com/p4lang/p4c/pull/4410)

Thank you for explanation sir.

>    I might work on this next week.

Great 😁

> FYI, there are certain names for these casing styles:
> 
> * CamelCase
> * camelBack
> * snake_case
> * kebab-case

Exactly that's my point, instead of writing 

>  _Variables are uncapitalized— e.g., `ipv4header`_

We might have to rephrase it as - `Snake-case` based on [Google Comment style guide](https://google.github.io/styleguide/cppguide.html#Variable_Names) or  `CamelCase` based on [this PR](https://github.com/p4lang/p4c/pull/4410#discussion_r1486393364)

Reply:
Given you clarified things, it should be okay to use snake_case for the time being? @fruffy

Reply:
I observe that camelCase is primarily used for the front end, and different naming formats as you mentioned above are used for the back end.

Reply:
This is not something I decide on my own. We first have to make sure other people agree before we do a refactoring. I will check this next week.

Reply:
Absolutely